### DISCUSSION TITLE
Is it possible to add labels to lines and polygons?

### DISCUSSION THREAD

**Discussion by frankrowe** [visgl/deck.gl#7100] [Q&A]
I am using a MVTLayer, but this would apply to the GeoJsonLayer as well.

It seems like the text accessors are only called for point features (getText, getPosition, etc). If it was called for other geometry types, I could calculate the position of the text (for example in the center of a polygon).

Is the way to do this by using Composite Layers? If so, how do you calculate the position of the text on a Line geometry? Is there a simpler way to do labels in general?

Thanks for any help!

**Comment by Pessimistress** [comment]
You can listen to `onTileLoad` and insert custom point features for the line strings/polygons.

To calculate label position/angle on the client side, see https://codepen.io/Pessimistress/pen/OJgmXba

However, I recommend moving the feature processing to the server side and embed it in your MVT files.
